clients/kafka-common/src/main/java/org/eclipse/hono/kafka/client/test/FakeProducer.java

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.kafka.client.test;
```

Comment:
do we really need this in the main source tree or can this go into the test folder?

Reply:
FMPOV this adds a missing _feature_ in the Vert.x Kafka client. The (Apache) Kafka client ships with a great class `MockProducer`, which cannot be used in Vert.x's client.
I see it as a _fake_ in the sense of https://martinfowler.com/bliki/TestDouble.html. Although it is used in test scenarios, it is a working implementation, which itself needs to be tested. Also, in practice, if I move it to the test folder, I cannot use it in other modules.

Reply:
Maybe this could go into the "test-utils" module. WDYT?

Reply:
Is this class required by other modules than the adapter-kafka module?

Reply:
I think this is useful for all unit tests involving a Kafka client. I expect to have at least one module for a client for backend applications (something like a module _application-kafka_ inside the module _clients_). And then there will probably be a Kafka client in the _command routing component_ discussed in #2273.

Reply:
Then you should indeed move it to the test-utils module

Reply:
@sophokles73 I now found a way to use the MockProducer of the Kafka client in the client of Vert.x. Therefore I removed the FakeProducer after all.